### ISSUE TITLE
Header values in quotes lose their quotes

### ISSUE THREAD

**Issue by efasel** [http4k/http4k#1023]
[ETag](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag) and [If-None-Match](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-None-Match) are headers where the values are quoted:
> It is a string of ASCII characters placed between double quotes, like "675af34563dc-tr34"

These quotes are lost here as the `headers.getCSV` method is called with `keepQuotes = false`: 
https://github.com/http4k/http4k/blob/457b8d70053454d39b5d9146c1a2f859820a1b0e/http4k-server/jetty/src/main/kotlin/org/http4k/server/Http4kJettyHttpHandler.kt#L35-L38

This later leads to a mismatch when comparing the if-none-match header value with the generated etag value.

This prevents us from upgrading to Jetty 12.

Co-Found-By @dhs3000

**Comment by efasel** [comment]
```kotlin
Subject: [PATCH] Add test for header values with quotes
---
Index: http4k-server/jetty/src/test/kotlin/org/http4k/server/Http4kJettyHttpHandlerTest.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/http4k-server/jetty/src/test/kotlin/org/http4k/server/Http4kJettyHttpHandlerTest.kt b/http4k-server/jetty/src/test/kotlin/org/http4k/server/Http4kJettyHttpHandlerTest.kt
new file mode 100644
--- /dev/null	(date 1704475064584)
+++ b/http4k-server/jetty/src/test/kotlin/org/http4k/server/Http4kJettyHttpHandlerTest.kt	(date 1704475064584)
@@ -0,0 +1,322 @@
+package org.http4k.server
+
+import com.natpryce.hamkrest.assertion.assertThat
+import com.natpryce.hamkrest.equalTo
+import org.eclipse.jetty.http.HttpField
+import org.eclipse.jetty.http.HttpFields
+import org.eclipse.jetty.http.HttpURI
+import org.eclipse.jetty.http.HttpVersion
+import org.eclipse.jetty.io.Connection
+import org.eclipse.jetty.io.Content
+import org.eclipse.jetty.server.Components
+import org.eclipse.jetty.server.ConnectionMetaData
+import org.eclipse.jetty.server.Connector
+import org.eclipse.jetty.server.Context
+import org.eclipse.jetty.server.HttpConfiguration
+import org.eclipse.jetty.server.HttpStream
+import org.eclipse.jetty.server.Session
+import org.eclipse.jetty.server.TunnelSupport
+import org.eclipse.jetty.util.Callback
+import org.http4k.client.ApacheClient
+import org.http4k.core.Body
+import org.http4k.core.Headers
+import org.http4k.core.HttpHandler
+import org.http4k.core.MemoryBody
+import org.http4k.core.MemoryRequest
+import org.http4k.core.Request
+import org.http4k.core.Response
+import org.http4k.core.Status
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.fail
+import java.io.InputStream
+import java.net.InetSocketAddress
+import java.net.SocketAddress
+import java.nio.ByteBuffer
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.TimeoutException
+import java.util.function.Consumer
+import java.util.function.Function
+import java.util.function.Predicate
+import java.util.function.Supplier
+
+class Http4kJettyHttpHandlerTest : ServerContract({ Jetty(it, ServerConfig.StopMode.Immediate) }, ApacheClient()) {
+    override fun requestScheme() = equalTo("http")
+
+    @Test
+    fun `header with quotes keep their quotes`() {
+        Http4kJettyHttpHandler(httpHandler).handle(request, response, callback)
+    }
+
+    private object httpHandler : HttpHandler {
+        override fun invoke(request: Request): Response {
+            if (request is MemoryRequest) {
+                val headerValue = request.headerValues("if-none-match").first()
+                assertThat(headerValue, equalTo("\"900150983cd24fb0d6963f7d28e17f72\""))
+            } else {
+                fail(RuntimeException("Could not assert"))
+            }
+            return response2
+        }
+    }
+
+    private object response2 : Response {
+        override val status: Status
+            get() = Status.NOT_MODIFIED
+
+        override fun header(name: String, value: String?): Response {
+            TODO("Not yet implemented")
+        }
+
+        override fun headers(headers: Headers): Response {
+            TODO("Not yet implemented")
+        }
+
+        override fun replaceHeader(name: String, value: String?): Response {
+            TODO("Not yet implemented")
+        }
+
+        override fun replaceHeaders(source: Headers): Response {
+            TODO("Not yet implemented")
+        }
+
+        override fun removeHeader(name: String): Response {
+            TODO("Not yet implemented")
+        }
+
+        override fun removeHeaders(prefix: String): Response {
+            TODO("Not yet implemented")
+        }
+
+        override fun body(body: Body): Response {
+            TODO("Not yet implemented")
+        }
+
+        override fun body(body: String): Response {
+            TODO("Not yet implemented")
+        }
+
+        override fun body(body: InputStream, length: Long?): Response {
+            TODO("Not yet implemented")
+        }
+
+        override fun status(new: Status): Response {
+            TODO("Not yet implemented")
+        }
+
+        override val headers: Headers
+            get() = emptyList()
+
+        override val body: Body
+            get() = MemoryBody("body")
+        override val version: String
+            get() = TODO("Not yet implemented")
+
+    }
+
+    private object request : org.eclipse.jetty.server.Request {
+        override fun removeAttribute(p0: String?): Any {
+            TODO("Not yet implemented")
+        }
+
+        override fun setAttribute(p0: String?, p1: Any?): Any {
+            TODO("Not yet implemented")
+        }
+
+        override fun getAttribute(p0: String?): Any {
+            TODO("Not yet implemented")
+        }
+
+        override fun getAttributeNameSet(): MutableSet<String> {
+            TODO("Not yet implemented")
+        }
+
+        override fun read(): Content.Chunk {
+            TODO("Not yet implemented")
+        }
+
+        override fun demand(demandCallback: Runnable?) {
+            TODO("Not yet implemented")
+        }
+
+        override fun fail(p0: Throwable?) {
+            TODO("Not yet implemented")
+        }
+
+        override fun getId(): String {
+            TODO("Not yet implemented")
+        }
+
+        override fun getComponents(): Components {
+            TODO("Not yet implemented")
+        }
+
+        override fun getConnectionMetaData(): ConnectionMetaData = testConnectionMetaData
+
+        override fun getMethod() = "GET"
+
+        override fun getHttpURI() = HttpURI.from("http://localhost")
+
+        override fun getContext(): Context {
+            TODO("Not yet implemented")
+        }
+
+        override fun getHeaders() = fields
+
+        override fun getTrailers(): HttpFields {
+            TODO("Not yet implemented")
+        }
+
+        override fun getBeginNanoTime(): Long {
+            TODO("Not yet implemented")
+        }
+
+        override fun getHeadersNanoTime(): Long {
+            TODO("Not yet implemented")
+        }
+
+        override fun isSecure(): Boolean {
+            TODO("Not yet implemented")
+        }
+
+        override fun consumeAvailable(): Boolean {
+            TODO("Not yet implemented")
+        }
+
+        override fun addIdleTimeoutListener(onIdleTimeout: Predicate<TimeoutException>?) {
+            TODO("Not yet implemented")
+        }
+
+        override fun addFailureListener(onFailure: Consumer<Throwable>?) {
+            TODO("Not yet implemented")
+        }
+
+        override fun getTunnelSupport(): TunnelSupport {
+            TODO("Not yet implemented")
+        }
+
+        override fun addHttpStreamWrapper(wrapper: Function<HttpStream, HttpStream>?) {
+            TODO("Not yet implemented")
+        }
+
+        override fun getSession(create: Boolean): Session {
+            TODO("Not yet implemented")
+        }
+
+    }
+
+    private object response : org.eclipse.jetty.server.Response {
+
+        var responseStatus = 200
+
+        override fun write(last: Boolean, byteBuffer: ByteBuffer?, callback: Callback?) {
+            byteBuffer?.clear()
+            println("last: $last")
+            callback?.succeeded()
+        }
+
+        override fun getRequest(): org.eclipse.jetty.server.Request {
+            TODO("Not yet implemented")
+        }
+
+        override fun getStatus() = responseStatus
+
+        override fun setStatus(code: Int) {
+            responseStatus = code
+        }
+
+        override fun getHeaders(): HttpFields.Mutable {
+            TODO("Not yet implemented")
+        }
+
+        override fun getTrailersSupplier(): Supplier<HttpFields> {
+            TODO("Not yet implemented")
+        }
+
+        override fun setTrailersSupplier(trailers: Supplier<HttpFields>?) {
+            TODO("Not yet implemented")
+        }
+
+        override fun isCommitted(): Boolean {
+            TODO("Not yet implemented")
+        }
+
+        override fun hasLastWrite(): Boolean {
+            TODO("Not yet implemented")
+        }
+
+        override fun isCompletedSuccessfully(): Boolean {
+            TODO("Not yet implemented")
+        }
+
+        override fun reset() {
+            TODO("Not yet implemented")
+        }
+
+        override fun writeInterim(status: Int, headers: HttpFields?): CompletableFuture<Void> {
+            TODO("Not yet implemented")
+        }
+
+    }
+
+    private object callback : Callback
+
+    private object fields : HttpFields {
+        override fun listIterator(index: Int): ListIterator<HttpField> {
+            return listOf(HttpField("if-none-match", "\"900150983cd24fb0d6963f7d28e17f72\"")).listIterator()
+        }
+
+    }
+
+    private object testConnectionMetaData : ConnectionMetaData {
+        override fun removeAttribute(p0: String?): Any {
+            TODO("Not yet implemented")
+        }
+
+        override fun setAttribute(p0: String?, p1: Any?): Any {
+            TODO("Not yet implemented")
+        }
+
+        override fun getAttribute(p0: String?): Any {
+            TODO("Not yet implemented")
+        }
+
+        override fun getAttributeNameSet(): MutableSet<String> {
+            TODO("Not yet implemented")
+        }
+
+        override fun getId(): String {
+            TODO("Not yet implemented")
+        }
+
+        override fun getHttpConfiguration(): HttpConfiguration {
+            TODO("Not yet implemented")
+        }
+
+        override fun getHttpVersion(): HttpVersion {
+            TODO("Not yet implemented")
+        }
+
+        override fun getProtocol(): String {
+            TODO("Not yet implemented")
+        }
+
+        override fun getConnection(): Connection {
+            TODO("Not yet implemented")
+        }
+
+        override fun getConnector(): Connector {
+            TODO("Not yet implemented")
+        }
+
+        override fun isPersistent(): Boolean {
+            TODO("Not yet implemented")
+        }
+
+        override fun getRemoteSocketAddress() = InetSocketAddress(80)
+
+        override fun getLocalSocketAddress(): SocketAddress {
+            TODO("Not yet implemented")
+        }
+
+    }
+}
```

**Comment by efasel** [comment]
Thanks to @dhs3000 the corresponding PR #1027 now has a nice test!

@daviddenton Would be great to have a release coming soon including the fixes for #1022 and #1023, then we can upgrade to Jetty 12. :checkered_flag:
